R/src.R

```diff
@@ -0,0 +1,307 @@
+  ## way.
+
+  ## TODO: some control parameter here to tune 'instance' or not
+  ## through views, data, connection; easier once we have variable
+  ## interpolation in orderly3 configuration, plus a helper in the db
+  ## plugin.
+
+  migrate <- list(
+    src_migrate_description,
+    src_migrate_parameters,
+    src_migrate_global_resources,
+    src_migrate_resources,
+    src_migrate_depends,
+    src_migrate_artefacts,
+    src_migrate_db_views,
+    src_migrate_db_data,
+    src_migrate_db_connection,
+    src_migrate_packages,
+    src_migrate_sources,
+    src_migrate_script)
+
+  dat <- orderly:::orderly_recipe$new(name, cfg, TRUE)
+  code <- if (strict) "orderly3::orderly_strict_mode()" else character(0)
+  for (f in migrate) {
+    code <- add_section(code, f(cfg, dat))
+  }
+  list(code = code, script = dat$script)
+}
+
+
+src_migrate_description <- function(cfg, dat) {
+  if (is.null(dat$displayname) && is.null(dat$description) &&
+      is.null(dat$custom)) {
+    return(NULL)
+  }
+  args <- list(display = dat$displayname,
+               long = dat$description,
+               custom = dat$fields)
+  args <- args[!vapply(args, is.null, TRUE)]
+  args_str <- paste(sprintf("\n  %s = %s", names(args),
+                            vapply(args, deparse1, "", width.cutoff = 500)),
+                    collapse = ",")
+  sprintf("orderly3::orderly_description(%s)", args_str)
+}
+
+
+src_migrate_parameters <- function(cfg, dat) {
+  if (is.null(dat$parameters)) {
+    return(NULL)
+  }
+  ## TODO: there's some inconsistency here with pluralisation in
+  ## orderly3, might need some updating.
+  fmt <- "orderly3::orderly_parameters(%s)"
+  args <- sprintf("%s = %s", names(dat$parameters),
+                  vapply(dat$parameters, deparse, ""))
+  sprintf(fmt, paste(args, collapse = ", "))
+}
+
+
+src_migrate_global_resources <- function(cfg, dat) {
+  if (is.null(dat$global_resources)) {
+    return(NULL)
+  }
+  fmt <- "orderly3::orderly_global_resource(%s)"
+  args <- sprintf('%s = "%s"',
+                  names(dat$global_resources), unname(dat$global_resources))
+  sprintf(fmt, paste(args, collapse = ", "))
+}
+
+
+src_migrate_resources <- function(cfg, dat) {
+  if (is.null(dat$resources)) {
+    return(NULL)
+  }
+  fmt <- "orderly3::orderly_resource(%s)"
+  if (length(dat$resources) == 1L) {
+    args <- dquote(dat$resources)
+  } else {
+    args <-sprintf("c(%s)", paste(dquote(dat$resources), collapse = ", "))
+  }
+  sprintf(fmt, args)
+}
+
+
+src_migrate_depends <- function(cfg, dat) {
+  if (is.null(dat$depends)) {
+    return(NULL)
+  }
+
+  ret <- character()
+  for (el in split(dat$depends, dat$depends$index)) {
+    name <- el$name[[1]]
+    query <- el$id[[1]]
+    there <- el$filename
+    here <- el$as
+    use <- sprintf("%s = %s", dquote_if_required(here), dquote(there))
+    if (length(there) == 1) {
+      str <- sprintf('orderly3::orderly_dependency("%s", "%s", c(%s))',
+                     name, query, use)
```

Comment:
If the query uses a parameter this is generating e.g.

```
orderly3::orderly_dependency("ago_data_areas", "latest(parameter:version == version)", c("depends/ago_areas.geojson" = "ago_areas.geojson"))
```

which doesn't work. Though I can't remember how are we supposed to pass in a parameter to the query and I can't see an example of that working in orderly3

Reply:
https://github.com/mrc-ide/inference-data/blob/main/src/aaa_areas_pull/orderly.yml#L33

Reply:
Right, so this needs to become `latest(parameter:version == this:version)` I think...